libdnf5/base/goal.cpp

```diff
@@ -3296,12 +3296,17 @@ void Goal::add_redo_transaction(
 void Goal::reset() {
     p_impl->module_specs.clear();
     p_impl->rpm_specs.clear();
+    p_impl->rpm_reason_change_specs.clear();
     p_impl->rpm_ids.clear();
     p_impl->group_specs.clear();
     p_impl->rpm_filepaths.clear();
     p_impl->resolved_group_specs.clear();
     p_impl->resolved_environment_specs.clear();
+    p_impl->group_specs.clear();
     p_impl->rpm_goal = rpm::solv::GoalPrivate(p_impl->base);
+    p_impl->serialized_transaction.reset();
+    p_impl->revert_transactions.reset();
+    p_impl->redo_transaction.reset();
```

Comment:
That's fine, but I was thinking why not we recreate the goal member in the daemon session. The goal object's destructor should handle cleanup automatically, right? That way, we wouldn't need to explicitly clear individual items here or worry about adding new ones in the future.

Reply:
We can do that in the daemon, no problem.
But since the `libdnf5::Goal::reset()` public API has already existed, I wanted to fix it. We can of course deprecate it and remove in the future.

Reply:
Let's keep it as it is now, I will create a new issue for that.

Reply:
https://github.com/rpm-software-management/dnf5/issues/1718